domain/schema/model.go

```diff
@@ -50,25 +51,130 @@ CREATE TABLE model_config (
 `)
 }
 
-func spacesSchema() schema.Patch {
+func spaceSchema() schema.Patch {
 	return schema.MakePatch(`
-CREATE TABLE provider_spaces (
+CREATE TABLE space (
     uuid            TEXT PRIMARY KEY,
-    name            TEXT
+    name            TEXT NOT NULL,
+    is_public       BOOLEAN
+);
+
+CREATE UNIQUE INDEX idx_spaces_uuid_name
+ON space (name);
+
+CREATE TABLE provider_space (
+    provider_id     TEXT PRIMARY KEY,
+    space_uuid      TEXT NOT NULL,
+    CONSTRAINT      fk_provider_space_space_uuid
+        FOREIGN KEY     (space_uuid)
+        REFERENCES      space(uuid)
+);
+
+CREATE UNIQUE INDEX idx_provider_space_space_uuid
+ON provider_space (space_uuid);
+`)
+}
+
+func subnetSchema() schema.Patch {
+	return schema.MakePatch(`
+CREATE TABLE subnet (
+    uuid                         TEXT PRIMARY KEY,
+    cidr                         TEXT NOT NULL,
+    vlan_tag                     INT,
+    is_public                    BOOLEAN,
+    space_uuid                   TEXT,
+    subnet_type_uuid             TEXT,
+    CONSTRAINT                   fk_subnets_spaces
+        FOREIGN KEY                  (space_uuid)
+        REFERENCES                   space(uuid)
+    CONSTRAINT                   fk_subnet_types
+        FOREIGN KEY                  (subnet_type_uuid)
+        REFERENCES                   subnet_type(uuid)
+);
+
+CREATE TABLE subnet_type (
+    uuid                         TEXT PRIMARY KEY,
+    name                         TEXT NOT NULL,
+    is_usable                    BOOLEAN,
+    is_space_settable            BOOLEAN
+);
+
+INSERT INTO subnet_type VALUES
+    (0, 'base', true, true),    -- The base (or standard) subnet type. If another subnet is an overlay of a base subnet in fan bridging, then the base subnet is the underlay in fan terminology.
+    (1, 'overlay', false, false),    
+    (2, 'overlay_segment', true, true);
+
+CREATE TABLE subnet_role_definition (
+    uuid                         TEXT PRIMARY KEY,
+    name                         TEXT NOT NULL
 );
 
-CREATE TABLE spaces (
+INSERT INTO subnet_role_definition VALUES
+    (0, 'overlay_of');    -- The subnet is an overlay of other (an underlay) subnet.
+
+CREATE TABLE subnet_type_role_mapping (
```

Comment:
Call this `subnet_type_association_type`. It is a reference table that declares allowable associations of one type to another.

It should have columns:
- `subject_subnet_type_uuid` (subnet type valid for association from).
- `associated_subnet_type_uuid` (subnet type valid for association to).
- `association_type_uuid` (fk to `subnet_association_type`).

Then you need another table `subnet_association`, which links subnets by association.
- `subject_subnet_uuid`
- `associated_subnet_uuid`
- `association_type_uuid`

Reply:
done, the only doubt I have is that I added the following index to the `subnet_association` table:
```
CREATE UNIQUE INDEX idx_subnet_association
ON subnet_association (subject_subnet_uuid, associated_subnet_uuid);
```
is it OK with you?

Reply:
I think that's OK.